src/Storage/Storage/File/Cmdlet/CloseAzureStorageFileHandle.cs

```diff
@@ -0,0 +1,294 @@
+                        break;
+
+                    case ShareNameCloseSingleParameterSetName:
+                    case ShareNameCloseAllParameterSetName:
+                        baseDirectory = this.BuildFileShareObjectFromName(this.ShareName).GetRootDirectoryReference();
+                        break;
+
+                    case ShareCloseSingleParameterSetName:
+                    case ShareCloseAllParameterSetName:
+                        baseDirectory = this.Share.GetRootDirectoryReference();
+                        break;
+                    case FileCloseAllParameterSetName:
+                        // Don't need to set baseDirectory when input is a CloudFile
+                        break;
+
+                    default:
+                        throw new PSArgumentException(string.Format(CultureInfo.InvariantCulture, "Invalid parameter set name: {0}", this.ParameterSetName));
+                }
+
+                if(ParameterSetName == ShareNameCloseSingleParameterSetName || ParameterSetName == ShareCloseSingleParameterSetName)
+                {
+                    this.Path = FileHandle.Path;
+                }
+
+                // When not input path/File, the list handle target must be a Dir
+                bool foundAFolder = true;
+                CloudFileDirectory targetDir = baseDirectory;
+                CloudFile targetFile = null;
+                if (this.File != null)
+                {
+                    targetFile = this.File;
+                    foundAFolder = false;
+                }
+                else
+                {
+                    if (!string.IsNullOrEmpty(this.Path))
+                    {
+                        string[] subfolders = NamingUtil.ValidatePath(this.Path);
+                        targetDir = baseDirectory.GetDirectoryReferenceByPath(subfolders);
+
+                        if (!targetDir.Exists())
+                        {
+                            foundAFolder = false;
+                        }
+
+                        if (!foundAFolder)
+                        {
+                            //Get file
+                            string[] filePath = NamingUtil.ValidatePath(this.Path, true);
+                            targetFile = baseDirectory.GetFileReferenceByPath(filePath);
+
+                            this.Channel.FetchFileAttributesAsync(
+                                targetFile,
+                                null,
+                                this.RequestOptions,
+                                this.OperationContext,
+                                this.CmdletCancellationToken).ConfigureAwait(false);
+                        }
+                    }
+                }
+
+                // Recursive only take effect on File Dir
+                if (!foundAFolder && Recursive.IsPresent)
+                {
+                    WriteWarning("The target object of the 'Path' is an Azure File, the parameter '-Recursive' won't take effect.");
```

Comment:
WriteWarning seems a bit excessive here, it isn't dangerous if it doesn't recurse.  I would suggest a Verbose message

Reply:
Fixed.